registry/client/repository.go

```diff
@@ -791,25 +947,33 @@ func (bs *blobs) Create(ctx context.Context, options ...distribution.BlobCreateO
 	if err != nil {
 		return nil, err
 	}
+	span.SetAttributes(attribute.String("URL", u))
+	span.SetAttributes(attribute.String("Method", http.MethodPost))
 
 	req, err := http.NewRequest("POST", u, nil)
```

Comment:
Doesn't this need to be changed to `NewRequestWithContext`? There are also some uses of `NewRequest` elsewhere in this file and in `registry/client/auth/session.go` and `registry/client/blob_writer.go`.

Reply:
What is the reason for changing to `NewRequestWithContext` here? If we are to achieve trace injection, I think it can be achieved by `otel.GetTextMapPropagator().Inject(traceCtx,propagation.HeaderCarrier(req.Header))`.

Reply:
I had thought that the transport would get the span information from the context, but if that's not how it works you can disregard the comment.

Reply:
You didn't understand it wrong,
1、Obtain the trace from the context and create a span, if there is one, create a new span if not
```go
span, traceCtx := tracing.StartSpan(ctx, "blobs.delete")
```
2、Inject trace into other http services, go through the following steps
```go
otel.GetTextMapPropagator().Inject(traceCtx, propagation.HeaderCarrier(req.Header))
```